File: src/main/java/tv/codealong/tutorials/various/audit/java/TransportRouteConfig.java
package tv.codealong.tutorials.various.audit.java;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.regex.Pattern;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TransportRouteConfig {
    private String field;
    private Pattern pattern;
    private String route;
    private boolean isDefault;

    protected boolean canEqual(Object other) {
        return other instanceof TransportRouteConfig;
    }
}

File: src/main/java/tv/codealong/tutorials/various/audit/TransportRouteConfig.kt
package tv.codealong.tutorials.various.audit

import java.util.regex.Pattern

data class TransportRouteConfig(
    val headerName: String,
    val pattern: Pattern,
    val topic: String,
    val isDefault: Boolean = false
) {
    init {
        require(headerName.isNotBlank()) { "Header name must not be blank" }
        require(topic.isNotBlank()) { "Topic must not be blank" }
    }
}